### PR TITLE
Bug 1732148: Properly documenting purpose of Image config.

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Adding further information to the API Image configuration struct.

### CODE REVIEW

File: config/v1/types_image.go [hunk 1]
```diff
@@ -6,7 +6,14 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
...
 
-// Image holds cluster-wide information about how to handle images.  The canonical name is `cluster`
+// Image config governs policies related to imagestream imports and runtime
+// configuration for external registries. It allows users to configure from
+// what registries OpenShift is allowed to import images from, to configure
+// extra CA trust bundles for these external registries among other configs.
+// When exposing OpenShift's image registry to external access, Image config
+// also governs what is the hostname used for the access.
+//
+// The canonical name is `cluster`.
```
Comment (adambkaplan) [inline config/v1/types_image.go]: Suggested edits:

```
// Image governs policies related to imagestream imports and runtime
// configuration for external registries. It allows cluster admins to configure which
// registries OpenShift is allowed to import images from, extra CA trust bundles 
// for external registries, and policies to blacklist/whitelist registry hostnames.
// When exposing OpenShift's image registry to the public, this also lets cluster admins
// specify the external hostname.
...
```

**Comment by adambkaplan** [comment]
@ricardomaraschini there is a `make update` command that syncs the godoc comments with other bits (like swagger) that are needed elsewhere.

File: config/v1/types_image.go (hunk 1 above)
Comment (ricardomaraschini) [reply config/v1/types_image.go]: Updated.

**Pull request merged**